flink-connector-kudu/src/main/java/org/apache/flink/connectors/kudu/table/KuduTableSource.java

```diff
@@ -109,12 +147,41 @@ public boolean isLimitPushedDown() {
         for (int i = 0; i < ints.length; i++) {
             fieldNames[i] = prevFieldNames.get(ints[i]);
         }
-        return new KuduTableSource(configBuilder, tableInfo, flinkSchema, fieldNames);
+        return new KuduTableSource(configBuilder, tableInfo, flinkSchema, bounded, predicates, fieldNames);
+    }
+
+    @Override
+    public TableSource<Row> applyPredicate(List<Expression> predicates) {
+        List<KuduFilterInfo> kuduPredicates = new ArrayList<>();
+        ListIterator<Expression> predicatesIter = predicates.listIterator();
+        while(predicatesIter.hasNext()) {
+            Expression predicate = predicatesIter.next();
+            KuduFilterInfo kuduPred = toKuduFilterInfo(predicate);
```

Comment:
Would be nicer to return Optional here but I leave this up to you

Reply:
Good suggestion, has changed to use `Optional` var